### ISSUE TITLE
Login / Name Storage Issue

### ISSUE THREAD

**Issue by zQueal** [marcwebbie/passpie#94]
I was replying to a mailing list and was required to create a `riseup.net` account. Because the password requirements are batshit crazy which required me to use a less than secure password, I saved the credentials into `Passpie` so I wouldn't forget them. My initial injection didn't work the way I wanted, so I ended up having to edit the record. As it stands now, my record looks like this;

    ├────────────┼──────────────────────┼────────────┼───────────┤
    │ riseup.net │ email@gmail.com      │ ********   │           │
    ╘════════════╧══════════════════════╧════════════╧═══════════╛

I'm trying to structure the `passpie copy` command, but I seem to not be able to. I tried `email@gmail.com@riseup.net` (which is how it's displayed in `passpie status`) but I get a credential not found error.

I tried simply `passpie copy riseup.net` however, upon entering my password and getting the `Password copied to clipboard` message I find that my clipboard is empty. I tested on other credentials and the CLI to clipboard is still working, so it's probably not a bug there.

Any ideas?

**Comment by marcwebbie** [comment]
Hello @zQueal 

What is credential `name` and `login`? 

if name = `riseup.net` and login = `email@gmail.com` then you can query the credential this way:

passpie email@gmail.com@riseup.net

However if you edited the credential file manually. You might want to check the values for `name`, `login` and `fullname` on the file. Also check the name of the file. It has to match the an expected path.

For example for an entry:

```
├────────────┼───────────────┼────────────┼───────────┤
│ riseup.net │ foo@gmail.com │ *****      │           │
╘════════════╧═══════════════╧════════════╧═══════════╛
```

The file path should be

```
/Users/username/.passpie
└── riseup.net
    └── foo@gmail.com.pass
```

**Comment by zQueal** [comment]
OK, so. I tried some things;

    λ passpie update riseup.net
    Name [riseup.net]:
    Login [zach.queal]: email@gmail.com
    Password [*****]:
    Repeat for confirmation:
    Comment []:

Which returns;

    ├────────────┼──────────────────────┼────────────┼───────────┤
    │ riseup.net │ email@gmail.com      │ ********   │           │
    ╘════════════╧══════════════════════╧════════════╧═══════════╛

So far, perfect! So I try to copy my password;

    λ passpie copy email@gmail.com@riseup.net
    Passphrase:
    Password copied to clipboard

After this, my clipboard is now empty--of which I also tried `passpie copy riseup.net` with the same result.

**Comment by marcwebbie** [comment]
Could you try adding as a new credential example?

```
passpie add foo+new@gmail.com@riseup.net --password "b4tsh1t cr4zy p4ssw0rd"
passpie copy foo+new@gmail.com@riseup.net
```

**Comment by zQueal** [comment]
This seems to be working flawlessly! 

You're not able to see it because I've censored it to try not to get spam in my email, however, my original email is `firstname.lastname@gmail.com`. Could the period in the middle be causing an issue?

**Comment by marcwebbie** [comment]
Let's keep it open to learn more what is happening. 

- What is your operating system?
- What is passpie installed version?

Could you try running:

```
export PASSPIE_DATABASE=tmp-passpie-issue && \
passpie --version && \
passpie init --passphrase s3cr3t && \
passpie add foo+new@gmail.com@riseup.net --force --password "b4tsh1t cr4zy p4ssw0rd" && \
passpie copy foo+new@gmail.com@riseup.net --to=stdout --passphrase s3cr3t && \
passpie && \
rm -rf tmp-passpie-issue
```

Output for me was:

```
passpie, version 1.4.0
Initialized database in tmp-passpie-issue
b4tsh1t cr4zy p4ssw0rd
╒════════════╤═══════════════════╤════════════╤═══════════╕
│ Name       │ Login             │ Password   │ Comment   │
╞════════════╪═══════════════════╪════════════╪═══════════╡
│ riseup.net │ foo+new@gmail.com │ ********   │           │
╘════════════╧═══════════════════╧════════════╧═══════════╛
```

**Comment by zQueal** [comment]
I'm running windows, so I'm unable to set env variables like that, so I just opted to use my own database. Hope it's what you were looking for;

    λ passpie --version
    passpie, version 1.4.0
    λ passpie add foo+new@gmail.com@riseup.net --force --password "b4tsh1t cr4zy p4ssw0rd"
    λ passpie copy foo+new@gmail.com@riseup.net --to=stdout
    Passphrase:
    b4tsh1t cr4zy p4ssw0rd
    λ passpie 
    ├────────────┼───────────────────────┼────────────┼───────────┤
    │ riseup.net │ foo+new@gmail.com     │ ********   │           │
    ├────────────┼───────────────────────┼────────────┼───────────┤

**Comment by marcwebbie** [comment]
Couldn't reproduce this on my box, it might be a windows issue. Will try to run on an windows box then

```
export PASSPIE_DATABASE=tmp-passpie-issue && \
passpie --version && \
passpie init --passphrase s3cr3t && \
passpie add foo.new@gmail.com@riseup.net --force --password "b4tsh1t cr4zy p4ssw0rd" && \
passpie copy foo.new@gmail.com@riseup.net --to=stdout --passphrase s3cr3t && \
passpie add foo...new@gmail.com@riseup.net --force --password "b4tsh1t cr4zy p4ssw0rd" && \
passpie copy foo...new@gmail.com@riseup.net --to=stdout --passphrase s3cr3t && \
passpie
rm -rf tmp-passpie-issue
```


Output:

```
passpie, version 1.4.1
Initialized database in tmp-passpie-issue
b4tsh1t cr4zy p4ssw0rd
b4tsh1t cr4zy p4ssw0rd
╒════════════╤═════════════════════╤════════════╤═══════════╕
│ Name       │ Login               │ Password   │ Comment   │
╞════════════╪═════════════════════╪════════════╪═══════════╡
│ riseup.net │ foo...new@gmail.com │ ********   │           │
├────────────┼─────────────────────┼────────────┼───────────┤
│ riseup.net │ foo.new@gmail.com   │ ********   │           │
╘════════════╧═════════════════════╧════════════╧═══════════╛
```

**Comment by bluefishblue** [comment]
I'm having the same issue as zQueal.  It seems the two bugs are

1)  When updating logins, you must now use the entire fullname (ex. passpie udpdate joe@test.com) where you used to just be able to use the name field (ex. passpie update test.com)

2)  If doing any kind of manual update (ex. passpie update joe@test.com), it loses the password and what gets copied to the clipboard is just blank

Here's how it looks so far:

- Not sure what version exactly this started happening, but I'm currently running 1.4.1 on OSX 10.11.3
- Creating new logins in passpie works fine, for example, joe@test.com.  If I create a new login, and then run passpie copy test.com command, the passwords gets copied to my clipboard and pastes properly
- If I take that same login and then run passpie update test.com (lets say I want to add a comment), it goes through the motions, but doesn't actually update anything
- If i run passpie update joe@test.com, it will make the changes and add my comment, but after doing this, when I run passpie copy test.com, the password says it is copied to the clipboard, but then is blank when trying to paste

**Comment by marcwebbie** [comment]
Hello @bluefishblue, @zQueal,

 this issue seems to be fixed in version 1.4.2 by commit: https://github.com/marcwebbie/passpie/commit/7d303677f371f38f8beb865da69ce8d46b44c6be

Please install latest version of `Passpie`and try again.

Thanks for your contribution

**Comment by marcwebbie** [comment]
I will be closing this issue for now, I am assuming it is fixed on latest version
